### PR TITLE
nfc: add debug PCAP output, refactor Mifare DESFire following #1095

### PR DESCRIPTION
**Pull request by kevinwallace**
# What's new

- When Settings > System > Debug is enabled, all NFC exchanges will be logged to `/ext/nfc/debug.pcap`.
- DESFire code has been refactored to use `furi_hal_nfc_tx_rx` instead of legacy `furi_hal_nfc_exchange`, which is now removed.  Follows in the footsteps of #1095.

# Verification 

- Enable debug mode, scan NFC tags, check `nfc/debug.pcap` on SD card.
- Disable debug mode, scan NFC tags, ensure behavior still correct and no PCAP written.

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by kevinwallace** [comment]
I initially thought this was ready for submission, but I'm working on some other changes on top of it and keep finding commits that should have been included in this PR (the fixups I've committed already, plus one more yet-to-be-separated-out change that fixes PICC<->PCD direction coding when emulating a PICC, as well as dealing with ISO14443 CRCs being offloaded to rfal and missing from the capture data).

Sorry about the noise.  I'm going to mark this as a draft, rebase/squash/cleanup, then re-submit for review.  General comments still welcome!

**Comment by kevinwallace** [comment]
OK, I think all the wrinkles have been ironed out.  Rebased on top of dev, and squashed those small fixups into a more coherent set of commits.  Ready for review again.

File: applications/nfc/helpers/nfc_debug_pcap.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <furi_hal_nfc.h>
+#include <storage/storage.h>
+
+// Prepare tx/rx context for debug pcap logging, if enabled.
```
Comment (gornekich) [inline applications/nfc/helpers/nfc_debug_pcap.h]: Could you rework comments in Doxygen style please

**Review by gornekich — COMMENTED** [review]
Great work!
Tested on "slow"  and "fast" SD cards, works fine

File: applications/nfc/helpers/nfc_debug_pcap.h (hunk 1 above)
Comment (kevinwallace) [reply applications/nfc/helpers/nfc_debug_pcap.h]: Done in a2ac64a9118236cd526a7f2fa337c098e2d88682

**Review by skotopes — APPROVED** [review]

**Pull request merged**